lib/src/core/guild/auto_moderation.dart

```diff
@@ -0,0 +1,305 @@
+}
+
+abstract class ITriggerMetadata {
+  /// Substrings wich will be searched for in the content.
+  /// The associated trigger type is [TriggerTypes.keyword].
+  List<String>? get keywordsFilter;
+
+  /// The internally pre-defined wordsets which will be searched for in content.
+  /// The associated trigger type is [TriggerTypes.keywordPreset].
+  List<KeywordPresets>? get keywordPresets;
+
+  /// Substrings which will be exempt from triggering the preset trigger type.
+  /// The associated trigger type is [TriggerTypes.keywordPreset].
+  List<String>? get allowList;
+
+  /// The total number of mentions (either role and user) allowed per message.
+  /// (Maximum of 50)
+  /// The associated trigger type is [TriggerTypes.mentionSpam]
+  // Pr still not merged
+  int? get mentionLimit;
+}
+
+abstract class IActionStructure {
+  /// The type of the action.
+  ActionTypes get actionType;
+
+  /// Additionnal metadata needed during execution for this specific action type.
+  IActionMetadata? get actionMetadata;
+}
+
+abstract class IActionMetadata {
+  /// The channel if to wich user content should be logged.
+  /// The associated action type is [ActionTypes.sendAlertMessage].
+  Cacheable<Snowflake, ITextGuildChannel>? get channelId;
+
+  /// The timeout duration - maximum duration is 4 weeks (2,419,200 seconds).
+  /// It's associated action type is [ActionTypes.timeout].
+  Duration? get duration;
+}
+
+class AutoModerationRule extends SnowflakeEntity implements IAutoModerationRule {
+  @override
+  late final Cacheable<Snowflake, IGuild> guild;
+
+  @override
+  late final String name;
+
+  @override
+  late final Cacheable<Snowflake, IMember> creator;
+
+  @override
+  late final EventTypes eventType;
+
+  @override
+  late final TriggerTypes triggerType;
+
+  @override
+  late final ITriggerMetadata triggerMetadata;
+
+  @override
+  late final List<IActionStructure> actions;
+
+  @override
+  late final bool enabled;
+
+  @override
+  late final List<RoleCacheable> ignoredRoles;
+
+  @override
+  late final List<ChannelCacheable<ITextGuildChannel>> ignoredChannels;
+
+  AutoModerationRule(RawApiMap rawData, INyxx client) : super(Snowflake(rawData['id'])) {
+    guild = GuildCacheable(client, Snowflake(rawData['guild_id']));
+    name = rawData['name'] as String;
+    creator = MemberCacheable(client, Snowflake(rawData['creator_id']), guild);
+    eventType = EventTypes._fromValue(rawData['event_type'] as int);
+    triggerType = TriggerTypes.fromValue(rawData['trigger_type'] as int);
+    triggerMetadata = TriggerMetadata(rawData['trigger_metadata'] as RawApiMap);
+    actions = [...?(rawData['actions'] as RawApiList?)?.map((a) => ActionStructure(a as RawApiMap, client))];
+    enabled = rawData['enabled'] as bool;
+    ignoredRoles = (rawData['exempt_roles'] as RawApiList).isNotEmpty
+        ? [...(rawData['exempt_roles'] as RawApiList).map((r) => RoleCacheable(client, Snowflake(r), guild))]
+        : [];
```

Comment:
thats overly complicated for now reason. You can just do `(rawData['exempt_roles'] as RawApiList).map((r) => RoleCacheable(client, Snowflake(r), guild))` and it will just work same as your code.

Reply:
Prefer calling `toList()` over `[...iterable]` for a single iterable as it is clearer what the intent of the code is.